rfd/0181-pg-database-access-webui.md

```diff
@@ -0,0 +1,322 @@
+
+Before her access, a system administrator had already enrolled a PostgreSQL
+instance and created a set of users the Teleport users could use. Those
+database users and the existent databases of the instance are listed on the
+role assigned to Alice.
+
+She first logs into Teleport's Web UI and then searches for the desired database
+on the resources list. After locating it, she clicks on the "Connect" button.
+
+![PostgreSQL instance resource card with connect button](assets/0181-connect-pg.png)
+
+After clicking, a modal window with connection information is presented. In that
+window, she needs to select which database and database user she'll be using.
+Teleport already fills this information based on her permissions, so she doesn't
+need to find this information somewhere else or ask someone. Also, this will
+prevent her from inputting the information incorrectly and being unable to
+connect.
+
+![PostgreSQL connect modal](assets/0181-connect-dialog.png)
+
+After selecting the required information, she's redirected to a new tab
+containing an interactive shell, similar to `psql`, where she can type
+her queries.
+
+![PostgreSQL interactive shell](assets/0181-pg-shell.png)
+
+After interacting with the database, she closes the tab, and her database
+session ends.
+
+##### Auto-user provisioning enabled
+
+This is the same scenario, but the PostgreSQL instance was configured with user
+provisioning enabled. This change implies which information Alice sees on the
+connect modal. She doesn’t need to select the database user, as it will default
+to her username. The select is then disabled, and a new now select will be
+presented where she can select database roles attached to their user.
+
+![PostgreSQL connect modal with database roles](assets/0181-connect-dialog-roles.png)
+
+#### User story: Bob access database with per-session MFA enabled.
+
+Bob is familiar with Teleport web UI and is experienced with PostgreSQL,
+spending considerable time using `psql`, the PostgreSQL CLI.
+
+The system administrators configured this Teleport cluster to use MFA per
+session (`require_session_mfa: true`). They also have granted Bob permission to
+connect to multiple PostgreSQL databases. The role he'll be using has a short
+session TTL.
+
+He logs into Teleport web UI, searches for the desired database, and clicks
+"Connect". He’s presented with a pre-filled form containing his connection
+information: database name, database roles, and database user. He confirms the
+values and clicks "Connect" to start a new session. He's then presented with a
+tab browser tab with the PostgreSQL interactive shell.
+
+In the new tab, he's prompted with the MFA modal (the same as SSH
+sessions). After completing the authentication, he starts performing multiple
+queries using the interactive shell, but after some time, he exceeds his session
+TTL, the connection is dropped, and he is redirected to the login page. His
+database session is terminated.
+
```

Comment:
I wrote a story like that but removed it. I hit some edge cases unrelated to database access in my user story. For example, when you request a database resource and the request is approved, it doesn't mean you can access it. This is due to your list of db_users or db_names needing to be paired with the database's needs.

Also, both scenarios connect as a single one, but the steps are independent. Once the user switches to the temporary role, they interact with Teleport web UI in the same way they do regularly.

> Also, what will be general impact of access requests on existing sessions?

Do you mean when the temporary permissions expire? If yes, the user got logged out, so it will be the same flow as when their session expires (covered in the MFA user story).